ic-cdk/src/call.rs

```diff
@@ -536,11 +532,17 @@ impl<T: AsRef<[u8]>> Future for CallFuture<T> {
                     ic0::call_perform()
                 };
 
-                // The conversion fails only when the err_code is 0, which means the call was successfully enqueued.
-                match CallPerformErrorCode::from(err_code) {
-                    CallPerformErrorCode::NoError => {}
-                    c => {
-                        let result = Err(CallError::CallPerformFailed(c));
+                match code {
+                    0 => {
+                        // call_perform returns 0 means the call was successfully enqueued.
+                    }
+                    _ => {
+                        let reject_code = RejectCode::try_from(code).unwrap();
+                        let result = Err(CallRejected {
+                            reject_code,
+                            reject_message: "failed to enqueue the call".to_string(),
```

Comment:
Could we define this as a string constant given that it is used at multiple places?